src/main/java/htsjdk/variant/vcf/AbstractVCFCodec.java

```diff
@@ -228,7 +228,21 @@ protected VCFHeader parseHeaderFromLines( final List<String> headerStrings, fina
         return this.header;
     }
 
-	/**
+    /**
+     * @return the header that was either explicitly set on this codec, or read from the file.
+     */
+    public VCFHeader getHeader() {
```

Comment:
What will happen if the header is modified out of the codec? Maybe it is better to return a copy of the header...

Reply:
I'm not sure if such caution is warrented as there's an api `setVCFHeader(VCFHeader, VCFHeaderVersion)` below which can overwrite these fields completely.

Reply:
I think that there is a difference in your use case: calling `setHeader` does not modify objects returned previously by the method; but if the header returned by `getHeader` is modified downstream, it affects the codec itself and objects retrieved previously. If I understand correctly the purpose of this PR, which is use in combination with disq, concurrent modification is possible. I think that either return a copy of the header (or document that the method is unsafe is needed, although I prefer the other option).

Reply:
Do the codecs mutate the header? I tried out the change here and it solved the problem: https://github.com/tomwhite/disq/commit/33622668f576f72aa50f4464734b4ccf68a617d9

Reply:
I see - thx. I think the codecs currently only do so to repair version inconsistencies (during parsing the header itself, and in setHeader) if `doOnTheFlyModifications` is on. So I don't see any problematic code path in your disq commit as it stands.

Reply:
@tomwhite Can you add a comment saying `Don't modify the returned value` and also that it can return null.

Reply:
@lbergelson Done. Thanks.